pkg/entity-store/postgres.go

```diff
@@ -0,0 +1,268 @@
+package entitystore
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/types"
+	_ "github.com/lib/pq"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+type postgres struct {
+	db *sqlx.DB
+}
+
+type dbEntity struct {
+	Key    string         `db:"key"`
+	Value  types.JSONText `db:"value"`
+	Status string         `db:"status"`
+}
+
+func (p *postgres) createTable() error {
+
+	sql := `
+		CREATE TABLE IF NOT EXISTS entity (
+		key 		TEXT PRIMARY KEY,
+		value 		JSONB,
+		status 		TEXT
+	)`
+	_, err := p.db.Exec(sql)
+	if err != nil {
+		return errors.Wrap(err, "fail to create the entity table")
+	}
+	return nil
+}
+
+func (p *postgres) dropTable() error {
+	sql := `
+	DROP TABLE IF EXISTS entity`
+	_, err := p.db.Exec(sql)
+	if err != nil {
+		log.Debug(err)
+		return errors.Wrap(err, "fail to drop the entity table")
+	}
+	return nil
+}
+
+func getHostAndPort(addr string) (string, string, error) {
+	res := strings.Split(addr, ":")
+	if len(res) != 2 {
+		return "", "", fmt.Errorf("Invalid Address")
+	}
+	return res[0], res[1], nil
+}
+
+func NewPostgres(config BackendConfig) (*postgres, error) {
+
+	opts := fmt.Sprintf("postgres://%s:%s@%s/%s?sslmode=disable", config.Username, config.Password, config.Address, config.Bucket)
+	log.Debugf("postgresql database options: %s", opts)
+	db, err := sqlx.Connect("postgres", opts)
+	if err != nil {
+		log.Debugf("error connecting to postgresql DB")
+		return nil, errors.Wrap(err, "Unable to connect to the postgres db server")
+	}
+	store := &postgres{db: db}
+
+	// create tables
+	err = store.dropTable()
```

Comment:
Are we dropping the table every time we start dispatch?  Shouldn't we persist?

Reply:
good catch, I will remove it